### PR TITLE
Add * to the end of flight mode string in CRSF

### PR DESCRIPTION
**Pull request by MingchenZhang**
Like how it is handled in betaflight and inav, adding a start char at the end of the flight mode string to indicate the craft is not armed. 
This allow the commonly used lua widget able to tell the correct arming status.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
I would find this very annoying, since armed status is presented in the OSD immediately after the flight mode abbreviation already

**Comment by MingchenZhang** [comment]
Oh, this won't affect the OSD, it just add a star at the end of the flight mode string send to the radio in telemetry. Actually similar to that OSD disarm sign, but on radio.

**Comment by Hwurzburg** [comment]
If this only affects the CRSF telem stream, and therefore only TX or GCS displays of mode, and not internal OSD display or MSP telem (for other OSDs including DJI/Fatshark OSDs) I have no objection if @yaapu  says if wont be a problem for passthrough in his scripts

**Comment by yaapu** [comment]
> If this only affects the CRSF telem stream, and therefore only TX or GCS displays of mode, and not internal OSD display or MSP telem (for other OSDs including DJI/Fatshark OSDs) I have no objection if @yaapu says if wont be a problem for passthrough in his scripts

No objections but since it's specific for a particular flight stack I'd put a bit somewhere to enable/disable it, perhaps RC_OPTIONS @andyp1per ?

**Review by andyp1per — CHANGES_REQUESTED** [review]
Add RC_OPTIONS bit to control

File: libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp [hunk 1]
```diff
@@ -921,7 +921,7 @@ void AP_CRSF_Telem::calc_flight_mode()
             sizeof(AP_CRSF_Telem::FlightModeFrame), 
             "%s%s", 
             notify->get_flight_mode_str(), 
-            hal.util->get_soft_armed() ? "" : "*"
+            !rc().use_crsf_lq_as_rssi() || hal.util->get_soft_armed() ? "" : "*"
```
Comment (Hwurzburg) [inline libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:924]: I dont think this is the correct option and logic

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp (hunk 1 above)
Comment (MingchenZhang) [reply libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp]: oops, copy and forgot, I just corrected. 
I think the logic is right, I just flipped it so it's now easier to read.

**Review by Hwurzburg — APPROVED** [review]

**Comment by Hwurzburg** [comment]
> Does the mission planner need to be updated? or the changed "@bitmask" comment is sufficient?

MP automatically acquires the info for dialog boxes from the metadata...no change needed

**Comment by MingchenZhang** [comment]
OK, Thanks for pointing out. 
@yaapu It seems you were requested to review. Could you approve this?

**Review by tridge — APPROVED** [review]

**Pull request merged**